ethdb/database.go

```diff
@@ -71,23 +71,26 @@ type KeyValueStore interface {
 type AncientReader interface {
 	// HasAncient returns an indicator whether the specified data exists in the
 	// ancient store.
-	HasAncient(kind string, number uint64) (bool, error)
+	HasAncient(typ string, kind string, number uint64) (bool, error)
```

Comment:
I don't get why we need both `typ` and `kind`?

Reply:
`typ` refers to the freezer. E.g. now we only have the `ancientChain` freezer for maintaining blockchain data. In the future it's possible to add more freezers, like reverse diffs and so on.

`typ` is used to route the freezer, `kind` is used to route the table in the freezer.